### PR TITLE
remove namespace field from scaleTargetRef for hpa

### PR DESCRIPTION
**Pull request by fivesheep**
Trying to enable hpa with this chart and getting the following error
```
unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(HorizontalPodAutoscaler.spec.scaleTargetRef): unknown field "namespace" in io.k8s.api.autoscaling.v2beta1.CrossVersionObjectReference
```

per `HorizontalPodAutoscaler` spec, it doesn't have this namespace field from `scaleTargetRef`, removing this line shall fix it. more example can be found from https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/#autoscaling-on-multiple-metrics-and-custom-metrics

### CODE REVIEW

**Comment by divyansh-gupta** [comment]
@fivesheep - Thank you for the pull request, we will review it soon. Can you please confirm that when you removed the line locally, HPA worked for you?

**Comment by fivesheep** [comment]
@divyansh-gupta yes, it works after deleting that line. however, the version of the hpa we have is a bit old. I am getting this error
```
Warning: autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2 HorizontalPodAutoscaler
horizontalpodautoscaler.autoscaling/aws-privateca-issuer created
```

i can update this from the same pr to upgrade this to autoscaling/v2

**Comment by bmsiegel** [comment]
That'd be great if you wouldn't mind adding that. Thanks!

**Comment by fivesheep** [comment]
@bmsiegel the pr has been updated and tested

**Comment by bmsiegel** [comment]
Yea I agree, I merged a fix to this recently (https://github.com/cert-manager/aws-privateca-issuer/commit/ccf4d2578f48be6f0a9a4b3bf167b462c0ca57d3). Would you mind rebasing?

**Comment by bmsiegel** [comment]
Thanks, running testing now

File: charts/aws-pca-issuer/templates/hpa.yaml [hunk 1]
```diff
@@ -14,20 +14,23 @@ spec:
...
   maxReplicas: {{ .Values.autoscaling.maxReplicas }}
   metrics:
     {{- if .Values.autoscaling.targetCPUUtilizationPercentage }}
     - type: Resource
       resource:
         name: cpu
-        targetAverageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
+        target:
+          type: Utilization
+          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
```
Comment (amibhi) [inline charts/aws-pca-issuer/templates/hpa.yaml:26]: This is backwards incompatible with `autoscaling/v2beta1` which is used in K8s <1.22. We will need to keep `targetAverageUtilization` for clusters running `autoscaling/v2beta1`, and use this for clusters running `autoscaling/v2`.

Comment (amibhi) [reply charts/aws-pca-issuer/templates/hpa.yaml:26]:
```
helm install issuer ./charts/aws-pca-issuer -n aws-privateca-issuer \
--set serviceAccount.create=false --set serviceAccount.name=aws-privateca-issuer-sa \
--set image.repository="localhost:5000/aws-privateca-issuer" --set image.tag=latest --set image.pullPolicy=Always \
--set autoscaling.enabled=true
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(HorizontalPodAutoscaler.spec.metrics[0].resource): unknown field "target" in io.k8s.api.autoscaling.v2beta1.ResourceMetricSource
make: *** [install-local] Error 1
```

**Comment by amibhi** [comment]
Made changes to use `targetAverageUtilization` if cluster is running `autoscaling/v2beta1`, otherwise use new format e.g.

```
        target:
          type: Utilization
          averageUtilization: {{ .Values.autoscaling.targetCPUUtilizationPercentage }}
```

I have tested this PR on a cluster running K8s v1.22.17 with `autoscaling/v2beta1` and a cluster running K8s v1.27.1 with `autoscaling/v2`, with results below

**v1.22.17**
```
[ec2-user@ip-10-0-0-41 aws-privateca-issuer]$ kubectl version
Client Version: v1.28.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.22.17
WARNING: version difference between client (1.28) and server (1.22) exceeds the supported minor version skew of +/-1

[ec2-user@ip-10-0-0-41 aws-privateca-issuer]$ helm install issuer ./charts/aws-pca-issuer -n aws-privateca-issuer \
--set serviceAccount.create=false --set serviceAccount.name=aws-privateca-issuer-sa \
--set image.repository="localhost:5000/aws-privateca-issuer" --set image.tag=latest --set image.pullPolicy=Always \
--set autoscaling.enabled=true
W1120 22:59:48.819326  269981 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2beta2 HorizontalPodAutoscaler
W1120 22:59:48.879265  269981 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2beta2 HorizontalPodAutoscaler
NAME: issuer
LAST DEPLOYED: Wed Nov 20 22:59:48 2024
NAMESPACE: aws-privateca-issuer
STATUS: deployed
REVISION: 1
TEST SUITE: None

[ec2-user@ip-10-0-0-41 aws-privateca-issuer]$ helm -n aws-privateca-issuer get manifest issuer
...
---
# Source: aws-privateca-issuer/templates/hpa.yaml
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  name: issuer-aws-privateca-issuer
  labels:
    helm.sh/chart: aws-privateca-issuer-v1.4.0
    app.kubernetes.io/name: aws-privateca-issuer
    app.kubernetes.io/instance: issuer
    app.kubernetes.io/version: "v1.4.0"
    app.kubernetes.io/managed-by: Helm
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: issuer-aws-privateca-issuer
  minReplicas: 1
  maxReplicas: 100
  metrics:
    - type: Resource
      resource:
        name: cpu
        targetAverageUtilization: 80
```

**v1.27.1**
```
ubuntu@ip-172-31-80-22:~/aws-privateca-issuer$ kubectl version
Client Version: v1.31.2
Kustomize Version: v5.4.2
Server Version: v1.27.1
WARNING: version difference between client (1.31) and server (1.27) exceeds the supported minor version skew of +/-1

ubuntu@ip-172-31-80-22:~/aws-privateca-issuer$ helm install issuer ./charts/aws-pca-issuer -n aws-privateca-issuer \
--set serviceAccount.create=false --set serviceAccount.name=aws-privateca-issuer-sa \
--set image.repository="localhost:5000/aws-privateca-issuer" --set image.tag=latest --set image.pullPolicy=Always \
--set autoscaling.enabled=true
NAME: issuer
LAST DEPLOYED: Wed Nov 20 22:52:13 2024
NAMESPACE: aws-privateca-issuer
STATUS: deployed
REVISION: 1
TEST SUITE: None

ubuntu@ip-172-31-80-22:~/aws-privateca-issuer$ helm -n aws-privateca-issuer get manifest issuer
...
---
# Source: aws-privateca-issuer/templates/hpa.yaml
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  name: issuer-aws-privateca-issuer
  labels:
    helm.sh/chart: aws-privateca-issuer-v1.4.0
    app.kubernetes.io/name: aws-privateca-issuer
    app.kubernetes.io/instance: issuer
    app.kubernetes.io/version: "v1.4.0"
    app.kubernetes.io/managed-by: Helm
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: issuer-aws-privateca-issuer
  minReplicas: 1
  maxReplicas: 100
  metrics:
    - type: Resource
      resource:
        name: cpu
        target:
          type: Utilization
          averageUtilization: 80
```

**Comment by amibhi** [comment]
Full testing worklog: [HPA worklog.md](https://github.com/user-attachments/files/17837878/HPA.worklog.md)

**Pull request merged**